velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -233,7 +233,14 @@ Writer::Writer(
   validateSchemaRecursive(schema_);
 
   if (options.flushPolicyFactory) {
-    flushPolicy_ = options.flushPolicyFactory();
+    std::unique_ptr<dwio::common::FlushPolicy> flushPolicy =
+        options.flushPolicyFactory();
+    std::unique_ptr<DefaultFlushPolicy> parquetFlushPolicy(
```

Comment:
To avoid the potential memory leak if dynamic cast throws.
```
std::unique_ptr<DefaultFlushPolicy> parquetFlushPolicy;
auto* flushPolicy = options.flushPolicyFactory().release();
try {
   parquetFlushPolicy.reset(dynamic_cast<>());
} catch (...) {
   delete flushPolicy;
}
```